exports/dataset.cpp

```diff
@@ -781,6 +803,11 @@ PYBIND11_MODULE(dataset, m) {
              return slice;
            },
            py::keep_alive<0, 1>())
+      .def("__setitem__",
+           [](VariableSlice &self, const std::tuple<Dim, py::slice> &index,
+              const VariableSlice &other) {
+             detail::pySlice(self, index).assign(other);
+           })
```

Comment:
Would be good to add __setitem__ for:
1. VariableSlice &self, const std::tuple<Dim, **gsl::index**> &index, const VariableSlice &other
2. Variable &self, const std::tuple<Dim, **gsl::index**> &index, const VariableSlice &other
then.

Reply:
did you mean i should do this as part of this PR or part of #152 ?

Reply:
@nvaytet, possibly, but if it has been already done for slices (ranges) here, a good idea is to have this also for indexes.

Reply:
yes it is a good idea, I added it in #152